extract auth_url from api key instead of config

SimonLab opened the pull request:
ref: https://github.com/dwyl/auth/issues/162

SimonLab commented on lib/token.ex:
```diff
@@ -29,10 +29,25 @@ defmodule AuthPlug.Token do
+  @doc """
+  `auth_url/0` returns the `auth_url` (the last part of the AUTH_API_KEY)
+  """
+  def auth_url do
+    api_key()
+    |> String.split("/")
+    |> List.last()
+    |> String.split("-")
+    |> List.last()
```
We could  use a regex here instead, a bit faster maybe but less readable.

SimonLab commented on lib/token.ex:
```diff
@@ -14,7 +14,7 @@ defmodule AuthPlug.Token do
    gives us the `client_id` and `client_secret`
    example:
-   2cfxNaWUwJBq1F4nPndoEHZJ5YCCNq9JDNAAR/2cfxNadrhMZk3iaT1L5k6Wt67c9ScbGNPz8Bw
+   2cfxNaWUwJBq1F4nPndoEHZJ5YCCNq9JDNAAR/2cfxNadrhMZk3iaT1L5k6Wt67c9ScbGNPz8Bw-https://dwylauth.herokuapp.com
```
Choose to use "-" to concatenate the auth url, see https://github.com/dwyl/auth/issues/162#issuecomment-968816648

SimonLab commented:
@nelsonic let me know if this what you had in mind with https://github.com/dwyl/auth/issues/162

nelsonic reviewed (changes requested):
@SimonLab this PR looks good. 🎉 
please see my comment on the issue: https://github.com/dwyl/auth/issues/162#issuecomment-968965202

SimonLab commented:
Thanks for the review @nelsonic I think we might have an issue with "|", see https://github.com/dwyl/auth/issues/162#issuecomment-968965202

nelsonic commented:
The **auth_url** should be just dwylauth.herokuapp.com (or eventually auth.dwyl.com in production)

SimonLab commented:
@nelsonic I've updated the PR to remove the `https://` form the environment variable. I've also updated the function to use a regex to find the different parts (splits by "/" now that "https://" is removed) as I don't think it is a too complicated one.

nelsonic commented on lib/token.ex:
```diff
@@ -14,25 +14,43 @@ defmodule AuthPlug.Token do
    gives us the `client_id` and `client_secret`
    example:
-   2cfxNaWUwJBq1F4nPndoEHZJ5YCCNq9JDNAAR/2cfxNadrhMZk3iaT1L5k6Wt67c9ScbGNPz8Bw
+   2cfxNaWUwJBq1F4nPndoEHZJ5YCCNq9JDNAAR/2cfxNadrhMZk3iaT1L5k6Wt67c9ScbGNPz8Bw-https://dwylauth.herokuapp.com
```
```suggestion
   2cfxNaWUwJBq1F4nPndoEHZJ5YCCNq9JDNAAR/2cfxNadrhMZk3iaT1L5k6Wt67c9ScbGNPz8Bw/dwylauth.herokuapp.com
```

nelsonic commented on mix.exs:
```diff
@@ -4,7 +4,7 @@ defmodule AuthPlug.MixProject do
     [
       app: :auth_plug,
-      version: "1.3.3",
+      version: "1.4.0",
```
I actually think this should be `2.0` given that the way we specify the `auth_url` in the `AUTH_API_KEY` is a breaking change. 💭 
Leaving it as `1.4` because we are basically the only ones _using_ this package right now. 
But if anyone `else` was using it, we'd have to do a proper SemVer release.

nelsonic reviewed (approved):
@SimonLab this looks superb. Thanks for humouring me. 🙌

nelsonic commented:
Package published to https://hex.pm/packages/auth_plug/1.4.0 📦 🚀
